File: src/main/java/com/goudourasv/http/instructors/dto/InstructorCreate.java
package com.goudourasv.http.instructors.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotBlank;
import java.util.List;
import java.util.UUID;

public class InstructorCreate {
    @NotBlank
    private String lastName;
    @NotBlank
    private String firstName;
    @JsonProperty("institutions")
    private List<UUID> institutionIds;


    public InstructorCreate(String firstName, String lastName, List<UUID> institutionIds) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.institutionIds = institutionIds;

    }

    public String getLastName() {
        return lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public List<UUID> getInstitutionIds() {
        return institutionIds;
    }
}

File: src/main/java/com/goudourasv/http/instructors/dto/InstructorCreate.kt
package com.goudourasv.http.instructors.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*
import javax.validation.constraints.NotBlank

data class InstructorCreate(
    @NotBlank val firstName: String,
    @NotBlank val lastName: String,
    @JsonProperty("institutions") val institutionsIds: List<UUID>
)
